### PR TITLE
scripts: check merge conflicts

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Conflicted codes are merged in to release branch.

### What is changed and how it works?

Check merge confliction before merging.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to cherry-pick to the release branch

### Release note

```release-note
None
```

### CODE REVIEW

File: scripts/check-merge-conflicts.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+result=$(find ./ -type f \( -iname \*.json -o -iname \*.sh \) | while read file_path; do
```
Comment (hi-rustin) [inline scripts/check-merge-conflicts.sh:1]: Won't our other files conflict? For example dockerfile?

Comment (overvenus) [reply scripts/check-merge-conflicts.sh]: Addressed.

**Comment by overvenus** [comment]
Check all tracked file takes about 0.6s.

```sh
time make check-merge-conflicts
check-merge-conflicts
make check-merge-conflicts  0.56s user 0.79s system 105% cpu 1.286 total
```

File: scripts/check-merge-conflicts.sh (hunk 1 above)
Comment (hi-rustin) [reply scripts/check-merge-conflicts.sh]: Thanks!

File: scripts/check-merge-conflicts.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+result=$(git ls-tree --full-tree --name-only -r HEAD | while read file_path; do
```
Comment (hi-rustin) [inline scripts/check-merge-conflicts.sh:1]: Do we need a license here?

**Review by hi-rustin — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

File: scripts/check-merge-conflicts.sh (hunk 2 above)
Comment (amyangfei) [reply scripts/check-merge-conflicts.sh:1]: In theory a license is required. However TiCDC doesn't check this in `scripts/check-copyright.sh` and most bash files also have no license provided.
Maybe we should add check and add missing license

File: scripts/check-merge-conflicts.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    if [[ `grep -E "^<<<<<<< HEAD$" "${file_path}"` ]]; then
+        echo "${file_path}"
+    fi
+done)
```
Comment (kennytm) [inline scripts/check-merge-conflicts.sh:5]: why not simply 

```sh
! git grep '^<<<<<<<'
```

🙃

File: scripts/check-merge-conflicts.sh (hunk 2 above)
Comment (hi-rustin) [reply scripts/check-merge-conflicts.sh]: > In theory a license is required. However TiCDC doesn't check this in `scripts/check-copyright.sh` and most bash files also have no license provided.
> Maybe we should add check and add missing license

Let me file an issue to land it.

**Pull request merged**